### DISCUSSION TITLE
Pi Zero I2c Display not working.

### DISCUSSION THREAD

**Discussion by Beforeeverythingends** [probonopd/MiniDexed#641] [Q&A] [answered]
Hi I recently made a minidexed for a friend using an old rpi 3b they had. That build went great. I used a 1602 display without i2c.

I liked it so much that I decided to buy a Pi Zero 2W and make one for myself. I haven't wired up an encoder or the DAC yet but I can't get the I2c 1602 display to work. 

It's not a contrast issue, the display does light up and show the white boxes. And the contrast adjustment on the I2c board works. 

I've changed the I2c address in the .ini file to 0x27 as per the instructions. 

Is there anything I'm missing? 

Though I'm not new to electronics in general I am very new to micro controllers. Was there something I was supposed to do with the pi Zero before installing mini dexed on the memory card?

Also just to rule out hardware faults I've used 3 memory cards and 2 displays.

Any help would be greatly appreciated. Thank you.

**Comment by probonopd** [comment]
Hi @Beforeeverythingends are you sure you have connected all wires correctly between the display and the Pi? Can you maybe take a picture showing how you wired things up?

**Reply by Beforeeverythingends** [reply]
![PXL_20240418_171713853](https://github.com/probonopd/MiniDexed/assets/164257422/c3b22ca5-94e8-49e6-aef2-16fe1421f287)
![PXL_20240418_171649333](https://github.com/probonopd/MiniDexed/assets/164257422/b4b0956b-b979-496f-b6e6-df97c2509534)


Hope those are clear enough. Thank you for replying so quickly.

**Reply by diyelectromusic** [reply]
Can't spot anything obviously wrong... Have you set the I2C address in the right part of the ini file - there is one for the HD44780 and a different setting for the SSD1306, so just double check you've set LCDI2CAddress=0x27 and that LCDEnabled=1.

Also, I think you're using a board with a PCF8574, which should be address 0x27, but there is a variant iirc that uses 0x3F too, but I thought that was the PCF8574A... but might be worth a try?

The usual behaviour of these modules is for the address pins to be default HIGH unless pulled LOW by jumpers, but I've seen expander modules with the opposite configuration (although not usually on LCDs), so it might be worth trying 0x20 (or 0x38) too, just in case...?

If you can boot the Pi into Raspbian there is a command line utility that shows you devices on the I2C bus.  If you have an Arduino, there is a I2C scanner sketch that will do the same thing...

Aside: Did you see the note on the hardware page about 5V vs 3Vs operation and level shifting and expanders?

It might be worth posting your minidexed.ini file here and a photo of the HDMI log for the Pi booting up.

Kevin

**Reply by Beforeeverythingends** [reply]
# HD44780 LCD
LCDEnabled=1
LCDPinEnable=17
LCDPinRegisterSelect=4
LCDPinReadWrite=0
LCDPinData4=22
LCDPinData5=23
LCDPinData6=24
LCDPinData7=25
LCDI2CAddress=0x27

That's from my .uni file. I forgot to say but I did try 0x3F before posting just in case. I'll give the other addresses a go when I get a chance. 
I've not used raspbian before but I'll try that and the hdmi log tomorrow when I have time.

I did see the bit about the level shift but I didn't realise it was relevant if I was using the 5v pin to power the display. 

Thank you for the replies.

**Reply by diyelectromusic** [reply]
It's a little tricky re level shifting... as it totally depends on the LCD board...

But so many people seem to build one without them, and it does seem to largely work for a lot of people... so I don't know... but to me, I'm pretty sure it could mean piping 5V into the I2C pins on the Pi, and I don't see how that could ever end well...

Kevin

**Reply by Beforeeverythingends** [reply]
Hi again. So I connected to a display via hdmi but I'm getting no signal. I changed the setting in the .ini:
MIDIDumpEnabled=1
ProfileEnabled=1

As far as I can tell I've done everything right. Does this mean I have a faulty pi Zero?

Thanks.

**Reply by Beforeeverythingends** [reply]
Ok so I installed an image of raspberry pi os and still get no signal. I think it's a faulty zero. Unless you all have any information in unaware of. 

Thanks

**Reply by diyelectromusic** [reply]
So you're getting no display at all on the Zero with a stock RPi OS and nothing on the GPIO pins?

If that's the case, I guess yes, you need to work out how to get the Pi working itself first following a standard Pi trouble shooting tutorial - things like power supply, proper connections, SD card issues, and so on.

Do you have a multimeter?  Are you seeing a good 3V3 and 5V on the GPIO for example?  I seem to recall there are some test points on a zero too...

MiniDexed will start printing to a HDMI display as soon as it powers up with no config changes required - assuming the files can be read on boot at all.

Kevin

**Reply by Beforeeverythingends** [reply]
Hi thanks for the reply. Nothing showed up on the hdmi at all. Screen would just show no signal. 

I tested the voltages on the pins right after the first time I connected the screen and it didn't work. They all seemed fine.

I've contacted pimoroni and I'll be returning this board and getting a replacement.

I appreciate all the help you've given. Hopefully the new one works fine. I'll update the thread when I get it and try again.

Thanks.

**Reply by probonopd** [reply]
Well, can you boot Raspberry Pi OS on this board?

**Reply by Beforeeverythingends** [reply]
Not received the new one yet. Raspberry pi os wouldn't boot on this one.

**Comment by Beforeeverythingends** [comment, marked as answer]
New one came. Everything seems to be working so far. 

Had a bit of confusion when it booted raspberry pi os but mini dexed wouldn't. 

Turned out that when I reformatted the SD card to install mini dexed the default file system was exfat. 

Changed it to fat32 and it booted mini dexed fine. I2c Display works both the LCD and the oled.

Thanks for all the help.

**Comment by Beforeeverythingends** [comment]
Got round to finishing the build tonight. Thanks for all the help and for this amazing project. 

![PXL_20240428_215929930](https://github.com/probonopd/MiniDexed/assets/164257422/f4d729d6-cd63-4c87-aff8-2856be26be5c)
![IMG-20240428-WA0004](https://github.com/probonopd/MiniDexed/assets/164257422/7ab5581e-533d-4e5d-939e-2a50dee8c8c1)
